client/src/components/Login.js

```diff
@@ -0,0 +1,48 @@
+import React, { useState } from 'react';
+import { Link } from 'react-router-dom';
+import axios from 'axios';
+
+export default function Login(props) {
+  // Better to use class component when dealing with many controlled inputs?
+  const [username, setUsername] = useState('');
+  const [password, setPassword] = useState('');
```

Comment:
You're able to use a custom hook to make even better use of state hooks:
```javascript
import { useState } from 'react';

const useFormInput = (type = 'text', initialValue = '') => {
  const [ value, setValue ] = useState(initialValue);

  return { value, setValue, type };
};

export default useFormInput;
```
and some of the possible usage:
```javascript
  const inputs = {
    username: useFormInput(),
    password: useFormInput('password')
  };
  
  const handleLogin = ev => {
    ev.preventDefault();
  
    // an example of making some of this stuff more programmatic
    for (let key in inputs) {
      if (inputs[key].value === '') {
        return props.displayError('Please fill out all fields.');
      }
    }
  
    const user = {
      username: inputs.username.value,
      password: inputs.password.value
    };
  
    axios
      .post('http://localhost:3030/api/login', user)
      .then(res => {
        localStorage.setItem('user-token', res.data.token);
        props.history.push('/users');
      }).catch(msg => (
        props.displayError(
          msg.response 
            ? msg.response.data.error 
            : 'There was an error while attempting login.'
        )
      ));
  
    // another option for a for loop
    Object.values(inputs).forEach(input => {
      input.setValue('');
    });
  };
  
  const handleChange = ev => {
    props.displayError(null);
    const { name, value } = ev.target;
    inputs[name].setValue(value);
  };
  
  return (
    <form onSubmit={handleLogin}>
      {Object.entries(inputs).map(([ name, input ]) => (
        <input
          type={input.type}
          placeholder={name[0].toUpperCase() + name.slice(1)}
          name={name}
          onChange={handleChange}
          value={input.value}
        />
      ))}
      <button>Sign In</button>
      <Link to="/signup">Not a user yet? Click here to sign up!</Link>
    </form>
  );
```